src/main/java/kr/hhplus/be/server/infrastructure/external/RedisLockManager.java

```diff
@@ -0,0 +1,39 @@
+package kr.hhplus.be.server.infrastructure.external;
+
+import org.springframework.data.redis.core.StringRedisTemplate;
+import org.springframework.data.redis.core.script.RedisScript;
+import org.springframework.stereotype.Component;
+
+import lombok.RequiredArgsConstructor;
+import java.time.Duration;
+import java.util.Collections;
+
+@Component
+@RequiredArgsConstructor
+public class RedisLockManager {
+    
+    private final StringRedisTemplate redisTemplate;
+    private static final Duration TIMEOUT = Duration.ofSeconds(5);
+
+    public boolean lock(String key, String value) {
```

Comment:
```suggestion
    public Mono<String> acquireLock(String key, Duration timeout) {
        String lockValue = UUID.randomUUID().toString();
        
        return Mono.fromCallable(() -> {
            Boolean acquired = redisTemplate.opsForValue()
                .setIfAbsent(key, lockValue, timeout);
            return Boolean.TRUE.equals(acquired) ? lockValue : null;
        })
        .subscribeOn(Schedulers.boundedElastic());
    }
```

이런식으로 리액티브 친화적으로 락을 하나 더 구현하셔야 할거같아요 ㅎㅎ